src/hooks/useObserver.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+
+/**
+ * UseObserver hook - This hook is used to observe the intersection of elements
+ * @returns an object containing the intersection entry data of the observed elements
+ */
```

Comment:
Is this hook used to know when an element enters/leaves the viewport? If yes, could you add it to the comment? If no, could you say in lay terms what it does? :smile:

Reply:
✅